### ISSUE TITLE
Train/Test/Eval data

### ISSUE THREAD

**Issue by emilienschultz** [emilienschultz/pyactivetigger#11]
Small question : for the moment, if I am not mistaken, there are two separate datasets to train the bert model, train and test, right ? But the test dataset is both used for the training (select the model) and calculate metrics (comparing predicted/truth).

We discussed about it but I can't remember what we said exactly.

What I would do is to use the annotated dataset, separate it in train/test, finetune the model, and then only use a truth dataset to check and calculate metrics.

Would it be ok ? 

@eollion @jboelaert

**Comment by jboelaert** [comment]
I would prefer having both measures: on the validation set (used to select the model) and on the test set (when present).
The validation set does not give a reliable measure of prediction error, but there is some virtue in using it I think, in order not to rely too heavily on the test set for model selection.

**Comment by emilienschultz** [comment]
I am not sure to be clear on what which data is which

The trainer object takes two dataset : train and eval.

Eval is the test set to evaluate the general quality of the model, right ? (it is not the test set used intermediary to fine tune the parameters, which I guess is done internally)

For the moment this is a different dataset, loaded by the user, right ?

Why is it not a dataset created before the training from the complete labelled dataset (like with a test_train_split method) ? To avoid variability ?

(I guess there is a reason).

(thanks for your previous answers, btw)

**Comment by jboelaert** [comment]
No pb for the answers, thank you for the questions!

The eval set given to the trainer is the one used to select the model, and is drawn randomly from the training set.
Several options in the UI: fraction of eval set (default .2), seed, whether to select model based on its best value, and the data exclusion options (minimum count, balance classes).

There is some virtue in not having a fixed eval dataset: not to overfit that small dataset. I would argue variability is a good thing here (though it might lead to overconfidence in shaky early results of course). Also, doing it like this is easier for the user that might not be interested in test error evaluation.

Also, there is some virtue in not using the final test dataset before the very end of model training and selection: having a reliable measure of error at the end, on data that was really not used before.

**Comment by jboelaert** [comment]
Yes, the random split is created beforehand, in shiny

**Comment by emilienschultz** [comment]
ok, now that's perfectly clear, and coherent with my common sense :)

Le ven. 26 janv. 2024 à 10:02, jboelaert ***@***.***> a
écrit :
